src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -46,6 +46,7 @@
 	private final Waiter executeDueWaiter;
 	private boolean enableImmediateExecution;
 	private final boolean pollAfterCompletion;
+	private boolean lastPollAcquiredAllAvailableExecutors;
```

Comment:
Could perhaps be made local to the method `executeDue()`? Might have to change to an AtomicBoolean in that case

Reply:
Sorry - I don't follow. Local as in: a local variable in the ```executeDue``` method?

Reply:
Yeah, sorry wrote it in a hurry. Declare the variable in the method in stead of in a field

Reply:
Since we need to track some data between polls (whether the last poll filled up our local work queue), I'm not sure what you suggest is possible.

Can you give me some (pseudo) code so I can attempt to rework this PR?

Reply:
Hmm, I did not realize we need to track the data between polls, what is the reason for that?

I was thinking to declare `lastPollAcquiredAllAvailableExecutors` in the method, something like this:
```
protected void executeDue() {
  ...
  AtomicBoolean lastPollAcquiredAllAvailableExecutors = false;
  ...
          .thenRun(() -> releaseExecutor(pickedExecution.get(), lastPollAcquiredAllAvailableExecutors));
  ...
```